pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -845,6 +845,7 @@ func buildSidecarInboundHTTPOpts(lb *ListenerBuilder, cc inboundChainConfig) *ht
 func (lb *ListenerBuilder) buildInboundNetworkFiltersForHTTP(cc inboundChainConfig) []*listener.Filter {
 	var filters []*listener.Filter
 
+	filters = append(filters, xdsfilters.IstioNetworkAuthenticationFilter)
```

Comment:
What is the problem this PR solves?

Reply:
https://github.com/istio/istio/issues/41585

Identity should not be assumed to come from TLS directly.